agent/functional_tests/util/utils.go

```diff
@@ -408,6 +404,84 @@ func (agent *TestAgent) resolveTaskDockerID(task *TestTask, containerName string
 	return "", errors.New("No containers matched given name")
 }
 
+func (agent *TestAgent) ConfirmTaskIsStoppedViaIntrospection(task *TestTask) (bool, error) {
+	var err error
+	var isStopped bool
+
+	for i := 0; i < 5; i++ {
+		isStopped, err = agent.confirmTaskIsStoppedViaIntrospection(task)
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return isStopped, err
+}
+
+func (agent *TestAgent) confirmTaskIsStoppedViaIntrospection(task *TestTask) (bool, error) {
```

Comment:
nit: seems that `confirmTaskIsStopped` and `confirmTaskIsRunning` can be combined into a single function with a `state` parameter.